code/modules/power/supermatter/supermatter_cascade_components.dm

```diff
@@ -52,10 +53,20 @@
 	if(!next_turf || locate(/obj/crystal_mass) in next_turf)
 		return
 
+	var/datum/component/supermatter_crystal/sm = GetComponent(/datum/component/supermatter_crystal)
```

Comment:
It's better but might still create some issues, mostly because it could be called up to 8 times from a single source. The best thing would be to store the datum as a var on the obj root and use that (I think at least)

Reply:
Store it on the initialize I mean

Reply:
Yeah, this is better than running it a dozen times on every process call, however it runs on every process call anyway